src/twisted/internet/defer.py

```diff
@@ -631,7 +631,7 @@ def chainDeferred(self, d: "Deferred[_DeferredResultT]") -> "Deferred[None]":
         d._chainedTo = self
         return self.addCallbacks(d.callback, d.errback)
 
-    def callback(self, result: _DeferredResultT) -> None:
+    def callback(self, result: Union[_DeferredResultT, Failure]) -> None:
```

Comment:
I did think it was odd to pass a `Failure` to `callback`, but that exists in Twisted.  Let me look again as to what code does that, but I don't think we can change the behavior of `callback` without breaking compatibility.

Reply:
Docstring updated.

We don't really have a choice about supporting that behavior; changing it would be a compatibility issue, and probably not a small one.